calc/calc.py

```diff
@@ -265,6 +265,11 @@ def eval_variable(x):
         return all_variables[casify(x[0])]
 
     def eval_function(x):
+        # This condition here is only for factorial function
+        if isinstance(x[1], numbers.Real):
```

Comment:
@ichintanjoshi this felt a little late since by now all the other evaluation has happened, what do you think about [this alternate proposal](https://github.com/openedx/openedx-calc/pull/103) where we fix the eval functions upstream instead?

Reply:
Yes @feanil  sure, that'd work

Reply:
@feanil  what should be the next course of actions ? 

As we discussed earlier, I agree with your changes and I think it's best to do it in that manner if they are appearing earlier in flow, instead of my approach.

How to proceed with those changes ? Should I change code in this branch ?

Reply:
How about you cherry-pick my commits onto your branch since you have more updates here and then we can land your PR and I'll close mine?